include/jemalloc/internal/log.h

```diff
@@ -84,12 +84,15 @@ if (config_log) {							\
 	}								\
 }
 
-#define log(log_var, format, ...)					\
+/* Call as log(log_var, "format_string %d", arg_for_format_string); */
+#define log(...)							\
 do {									\
-	log_do_begin(log_var)						\
-		malloc_printf("%s: " format "\n",			\
-		    (log_var).name, __VA_ARGS__);			\
-	log_do_end(log_var)						\
+	log_do_begin(JEMALLOC_VA_ARGS_HEAD(__VA_ARGS__))		\
+		malloc_printf("%s: ",					\
+		    (JEMALLOC_VA_ARGS_HEAD(__VA_ARGS__)).name);		\
+		malloc_printf(JEMALLOC_VA_ARGS_TAIL(__VA_ARGS__));	\
+		malloc_printf("\n");					\
```

Comment:
It's hard, because we'd have to insert some of our things in the middle of some of theirs, and we wouldn't know if we need a comma or not. It's hard to fix that portably and without a bunch of macro hacks.